lib/cube_browser/browse.py

```diff
@@ -0,0 +1,80 @@
+import glob
+import IPython.display
+import ipywidgets
+import iris
+import matplotlib.pyplot as plt
+
+import cube_browser
+
+# Clear output, such as autosave disable notification.
+IPython.display.clear_output()
+
+# Defines the file system path for input files.
+path = ipywidgets.Text(
+    description='Path:',
+    value=iris.sample_data_path('GloSea4'),
+)
+def handle_path(sender):
+    options = glob.glob('{}/*'.format(path.value))
+    options.sort()
+    files.options = options
+# path is the widget which i want to observe
+# the first argument is the function to run on 'event'
+# names is the thing that i want know about, within path
+path.observe(handle_path, names='value')
+
+# Use default path value to initialise file options.
+options = glob.glob('{}/*'.format(path.value))
+options.sort()
+
+# Defines the files selected to be loaded.
+files = ipywidgets.SelectMultiple(
+    description='Files:',
+    options=options,
+    width='100%'
+)
+
+# Load action.
+load_button = ipywidgets.Button(description="Load these Files!")
+def handle_load(sender):
+    cubes = iris.load(files.value)
+    cube_picker.options = dict([(cube.summary(shorten=True), cube) for cube in cubes])
+load_button.on_click(handle_load)
```

Comment:
I agree with lumping these bits together.  I think it makes it more clear that they relate to each other.

Personally, though, I am still slightly unsure about what 'sender' is.